src/crypto/mod.rs

```diff
@@ -3,32 +3,31 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 use crate::error::*;
+use std::any::Any;
 
-pub trait RemotePublicKey {
-    /// Import the key component in the
-    /// binary uncompressed point representation.
-    fn from_raw(raw: &[u8]) -> Result<Self>
-    where
-        Self: Sized;
+pub(crate) mod holder;
+#[cfg(feature = "backend-openssl")]
+mod openssl;
+
+#[cfg(not(feature = "backend-openssl"))]
+pub use holder::{set_boxed_cryptographer, set_cryptographer};
+
+pub trait RemotePublicKey: Send + Sync + 'static {
     /// Export the key component in the
     /// binary uncompressed point representation.
     fn as_raw(&self) -> Result<Vec<u8>>;
+    /// For downcasting purposes.
+    fn as_any(&self) -> &Any;
 }
 
-pub trait LocalKeyPair {
-    /// Generate a random local key pair.
-    fn generate_random() -> Result<Self>
```

Comment:
Here and below, removed the `Result<Self>` methods to make these traits object-safe.